prometheus-to-sd/config/source_config.go

```diff
@@ -96,10 +90,7 @@ func (config *SourceConfig) UpdateWhitelistedMetrics(list []string) {
 
 // SourceConfigsFromFlags creates a slice of SourceConfig's base on the provided flags.
 func SourceConfigsFromFlags(source flags.Uris, component *string, host *string, port *uint, whitelisted *string, podId *string, namespaceId *string) []SourceConfig {
-	podConfig := PodConfig{
-		PodId:       *podId,
-		NamespaceId: *namespaceId,
-	}
+	podConfig := NewPodConfig(*podId, *namespaceId, "", "", "")
 	var sourceConfigs []SourceConfig
 	for _, c := range source {
 		if sourceConfig, err := parseSourceConfig(c, podConfig); err != nil {
```

Comment:
You also need to extract all values in parseSourceConfig function above, otherwise you will not be able to configure it using --source. 

dynamic_source actually is going to be removed.

Reply:
done